runtime/vm/BytecodeInterpreter.hpp

```diff
@@ -7301,10 +7306,77 @@ done:;
 	VMINLINE bool
 	isSubstitutable(j9object_t lhs, j9object_t rhs)
```

Comment:
instead of passing j9object_t, you should pass in that start of the payload (so j9object shifted down by the size of the header). This willmake it easier to compare flattened types